src/bionics.cpp

```diff
@@ -1512,9 +1978,39 @@ bool player::install_bionics(const itype &type, int skill_level)
                                 difficult);
     }
 
-    if (!query_yn(
-            _("WARNING: %i percent chance of genetic damage, blood loss, or damage to existing bionics! Install anyway?"),
-            100 - chance_of_success)) {
+    std::string warning = string_format(
+_( "WARNING: %i%% chance of genetic damage, blood loss, or damage to existing bionics!" ),
+                                              100 - chance_of_success );
+    uimenu bp_selection;
+    bp_selection.return_invalid = true;
+    bp_selection.title = string_format( _( "Bionic Installation: %s" ),
+                                        bionics[bioid].name.c_str() );
+    bp_selection.text = string_format( _( "%s Choose body part for installation:" ),
+                                       warning.c_str() );
+    std::vector<body_part> available_bodyparts;
+    for( size_t i = 0; i < num_bp; ++i ) {
+        body_part bp = static_cast<body_part>( i );
+        if( get_free_bionics_slots( bp ) > 0 ) {
+            bp_selection.addentry( bp_asText[i] );
+            available_bodyparts.push_back( bp );
+        }
+    }
+    if( available_bodyparts.empty() ) {
+        popup( _( "Not enough space for bionic installation!" ) );
+        return false;
+    } else if( available_bodyparts.size() == 1 ) {
+        // only one option is possible
+        bp_selection.ret = 0;
+        if( !query_yn( string_format( _( "%s Continue anyway?" ), warning.c_str() ).c_str() ) ) {
```

Comment:
`query_yn` works with format strings, you don't need to format it yourself. Remove the `string_format` call and use its arguments for `query_yn` directly.